tasks/undercloud.yml

```diff
@@ -146,3 +164,224 @@
+  when: ipa_conf_stat.stat.exists
+
+- name: Set novajoin join.conf location
+  set_fact:
+    novajoin_conf_file: "{{ container_prefix }}/etc/novajoin/join.conf"
+
+- name: Verify that join.conf exists
+  stat:
+    path: "{{ novajoin_conf_file }}"
+  register: join_conf_st
+  tags:
+    - post-undercloud-deploy
+
+# We really can't continue without join.conf
+- name: Fail if join.conf is not present
+  fail:
+    msg: join.conf is necessary to continue the validations
+  when: not join_conf_st.stat.exists|bool
+  tags:
+    - post-undercloud-deploy
+
+# We need to lookup stuff from the novajoin join.conf file but its
+# only readable by root.  make a temporary copy here, and read everything
+# we need.
+
+- name: Copy join.conf
+  command: cp "{{ novajoin_conf_file }}" /tmp/temp_join.conf
+  tags:
+    - post-undercloud-deploy
+  become: true
+
+- name: Chown copy of join.conf
+  file:
+    path: /tmp/temp_join.conf
+    owner: "{{ ansible_user }}"
+  tags:
+    - post-undercloud-deploy
+  become: true
+
+- name: get needed facts from join.conf
+  set_fact:
+    novajoin_keytab_path: "{{ container_prefix}}/{{ lookup('ini', 'keytab section=DEFAULT file=/tmp/temp_join.conf') }}"
+    novajoin_server_port: "{{ lookup('ini', 'join_listen_port section=DEFAULT file=/tmp/temp_join.conf') }}"
+    novajoin_server_host: "{{ lookup('ini', 'join_listen section=DEFAULT file=/tmp/temp_join.conf') }}"
+
+- name: delete join.conf copy
+  file:
+    path: /tmp/temp_join.conf
+    state: absent
+  tags:
+    - post-undercloud-deploy
+
+- name: verify the novajoin keytab is usable
+  become: true
+  tags:
+    - post-undercloud-deploy
+  block:
+    - name: Check for novajoin kerberos host keytab
+      stat:
+        path: "{{ novajoin_keytab_path }}"
+      register: novajoin_krb5_keytab_stat
+
+    - name: Test if novajoin principal in novajoin keytab is usable
+      command: kinit -kt "{{ novajoin_keytab_path }}" -c /tmp/my_novajoin_krb5_ccache "{{ novajoin_principal }}"
+      become: true
+      register: novajoin_principal_usable_result
+      ignore_errors: true
+      when:
+        - novajoin_krb5_keytab_stat.stat.exists
+
+    - name: Set facts for novajoin principal is usable skipped
+      set_fact:
+        principal_usable_status: '{{ helper_status_skipped }}'
+        principal_usable_reason: "skipped checking if the novajoin principal is usable for host {{ ansible_host }} because there is no keytab file"
+        principal_usable_recommendations: null
+      when:
+        - not novajoin_krb5_keytab_stat.stat.exists
+
+    - name: Set facts for novajoin principal is usable success
+      set_fact:
+        principal_usable_status: '{{ helper_status_ok }}'
+        principal_usable_reason: "The principal {{ novajoin_principal }} is able to obtain a kerberos ticket"
+        principal_usable_recommendations: null
+        temp_krb_caches: "{{ temp_krb_caches + [ '/tmp/my_novajoin_krb5_ccache' ] }}"
```

Comment:
oho! I didn't know you could do this!

Reply:
yup  -- all variables are global, and handlers always run at the end.